tests/integration/test_agent.py

```diff
@@ -1,24 +1,27 @@
 import asyncio
 import os
+import shutil
 import subprocess
 
 import pytest
 
 from opendevin.core.main import main
 
+workspace_base = os.getenv('WORKSPACE_BASE')
+
 
-# skip if
 @pytest.mark.skipif(
     os.getenv('AGENT') == 'CodeActAgent'
     and os.getenv('SANDBOX_TYPE').lower() == 'exec',
     reason='CodeActAgent does not support exec sandbox since exec sandbox is NOT stateful',
 )
 def test_write_simple_script():
     task = "Write a shell script 'hello.sh' that prints 'hello'. Do not ask me for confirmation at any point."
-    asyncio.run(main(task))
+    controller = asyncio.run(main(task))
+    asyncio.run(controller.close())
```

Comment:
@rbren FYI: this is not the most elegant solution but is the best compromise I can come up with, to fix the tests. Otherwise the sandbox and stuff are not shutdown properly after a test's run.